File: app/src/main/java/com/example/foodappprm/adapters/CategoryAdapter.java
package com.example.foodappprm.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;
import com.example.foodappprm.R;
import com.example.foodappprm.model.Category;

import java.util.List;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder> {
    private final List<Category> categories;
    private final OnCategoryClickListener onCategoryClickListener;
    private int selectedPosition = -1; // -1 means no selection

    public interface OnCategoryClickListener {
        void onCategoryClick(Category category);
    }

    public CategoryAdapter(List<Category> categories, OnCategoryClickListener listener) {
        this.categories = categories;
        this.onCategoryClickListener = listener;
    }

    public void clearSelection() {
        int oldPosition = selectedPosition;
        selectedPosition = -1;
        notifyItemChanged(oldPosition);
    }

    public class CategoryViewHolder extends RecyclerView.ViewHolder {
        private ImageView iconImageView;
        private TextView nameTextView;

        public CategoryViewHolder(@NonNull View itemView) {
            super(itemView);
            iconImageView = itemView.findViewById(R.id.categoryIconImageView);
            nameTextView = itemView.findViewById(R.id.categoryNameTextView);
        }

        public void bind(final Category category, boolean isSelected) {
            nameTextView.setText(category.getName());

            int resourceId = itemView.getContext().getResources().getIdentifier(
                category.getIconName(),
                "drawable",
                itemView.getContext().getPackageName()
            );

            if (resourceId != 0) {
                iconImageView.setImageResource(resourceId);
            } else {
                iconImageView.setImageResource(R.drawable.placeholder_image);
            }

            itemView.setBackgroundResource(
                isSelected ? R.drawable.bg_category_selected : R.drawable.bg_category_normal
            );
        }
    }

    @NonNull
    @Override
    public CategoryViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_category, parent, false);
        return new CategoryViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryViewHolder holder, int position) {
        Category category = categories.get(position);
        holder.bind(category, position == selectedPosition);

        holder.itemView.setOnClickListener(v -> {
            int previousSelected = selectedPosition;
            selectedPosition = position;
            notifyItemChanged(previousSelected);
            notifyItemChanged(selectedPosition);
            onCategoryClickListener.onCategoryClick(category);
        });
    }

    @Override
    public int getItemCount() {
        return categories != null ? categories.size() : 0;
    }
}

File: app/src/main/java/com/example/foodappprm/adapters/CategoryAdapter.kt
package com.example.foodappprm.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.foodappprm.R
import com.example.foodappprm.model.Category

class CategoryAdapter(
    private val categories: List<Category>,
    private val onCategoryClick: (Category) -> Unit
) : RecyclerView.Adapter<CategoryAdapter.CategoryViewHolder>() {

    private var selectedPosition = -1 // -1 means no selection

    fun clearSelection() {
        val oldPosition = selectedPosition
        selectedPosition = -1
        notifyItemChanged(oldPosition)
    }

    inner class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val iconImageView: ImageView = itemView.findViewById(R.id.categoryIconImageView)
        private val nameTextView: TextView = itemView.findViewById(R.id.categoryNameTextView)

        fun bind(category: Category, isSelected: Boolean) {
            nameTextView.text = category.name

            // Load icon từ drawable sử dụng resource identifier
            val resourceId = itemView.context.resources.getIdentifier(
                category.iconName,
                "drawable",
                itemView.context.packageName
            )

            if (resourceId != 0) {
                iconImageView.setImageResource(resourceId)
            } else {
                iconImageView.setImageResource(R.drawable.placeholder_image)
            }

            // Set background based on selection state
            itemView.setBackgroundResource(
                if (isSelected) R.drawable.bg_category_selected
                else R.drawable.bg_category_normal
            )

            // Update text color based on selection
            nameTextView.setTextColor(
                itemView.context.getColor(
                    if (isSelected) R.color.white
                    else R.color.black
                )
            )

            itemView.setOnClickListener {
                if (adapterPosition != selectedPosition) {
                    val oldPosition = selectedPosition
                    selectedPosition = adapterPosition
                    notifyItemChanged(oldPosition)
                    notifyItemChanged(selectedPosition)
                    onCategoryClick(category)
                }
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_category, parent, false)
        return CategoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
        holder.bind(categories[position], position == selectedPosition)
    }

    override fun getItemCount() = categories.size
}
